oak_containers_sdk/src/crypto.rs

```diff
@@ -130,3 +148,31 @@ impl AsyncRecipientContextGenerator for InstanceEncryptionKeyHandle {
         Ok(crypto_context)
     }
 }
+
+#[async_trait(?Send)]
+pub trait Signer {
```

Comment:
I think the trait could be moved to `oak_crypto`. So that SDK only have implementations

Reply:
Do we want clients to rely on the trait in oak_crypto? And do we want to commit to sharing that trait between oak containers and oak restricted kernel? That seems unnecessary IMO, in fact it may be easier to keep the SDKs separate as much as possible.
Though from the client library point of view we want to be able to verify attestations (and signatures, and do encryption) for both with the same library. Is that a fair high level goal?

Reply:
We definitely need to have `trait SigningKeyHandle` in `oak_crypto`, but as for the `Signer` I'm not sure (as an example, we can define it in `oak_crypto` and then just `pub use` it in SDKs)

Reply:
Why does `SigningKeyHandle` need to be in `oak_crypto`? That's only if we are planning to make the two SDKs match exactly, but I thought that's a non-goal -- or is it a goal?

Reply:
It's preferable to have SDKs be very similar.

I now realized that we don't have key handles for signing (only for encryption), so yes, maybe it's ok to have `trait Signer` redefined in each SDK. But personally I would prefer to define it in `oak_crypto` and then just `pub use` it in SDKs (so that we don't have unnecessary code duplication even if it's very small).

But it's not a strong opinion, so up to the preference

Reply:
> I think the trait could be moved to `oak_crypto`. So that SDK only have implementations

impossible. `oak_crypto` is no_std, the async_trait macro uses std.